interface/models.py

```diff
@@ -94,7 +94,11 @@ def url_for(self, filename):
         branch = self.repo_branch or "master"
         path = f"{self.repo_path}/" if self.repo_path else ""
         return urljoin(url_base, f"{branch}/{path}{filename}")
-
+    
+    def refresh_submission_penalty(self):
+        for submission in self.submission_set.all():
+            submission.penalty = None
+            submission.save()
```

Comment:
I'm using it this way (instead of calling a single update) to trigger the signal that updates the submission.